SConstruct

```diff
@@ -251,34 +257,40 @@ if not env['LIBPATH']:
         Abort('ROOT libraries could not be found.')
 
 
-    ## HDF5 configuration ----------------------------------
-
-    if env['HDF5_DIR'] != NULL_PATH:
-        env.PrependENVPath('PATH', env['HDF5_DIR'])
-    try:
-        env['HDF5_LIB'] = os.environ['HDF5_LIB']
-        env.Append( LIBPATH = [env['HDF5_LIB']] )
-        env.Append(LIBS = ['hdf5'])
-        env['HDF5_INC'] = os.environ['HDF5_INC']
-        env.Append( CPPPATH = [env['HDF5_INC']] )
-    except KeyError: pass
-
-
-    ## GSL configuration --------------------------   -------
+    ## GSL configuration ------------------------------------
 
-    if env['GSL_DIR'] != NULL_PATH:
+    if env['GSL_BINDIR'] != NULL_PATH:
         env.PrependENVPath('PATH', env['GSL_DIR'])
 
     env.ParseConfig('gsl-config --cflags --libs')
 
     if not conf.CheckCXXHeader('gsl/gsl_errno.h'):
         Abort('GSL headers not found.')
 
+    if not conf.CheckLib(library='gsl', language='CXX', autoadd=0):
+        Abort('GSL library not found.')
+
  #   env.Append(LIBS = ['gsl','gslcblas'])
 
 
-#    if not conf.CheckLib(library='GSL', language='CXX', autoadd=0):
-#        Abort('GSL library not found.')
+    ## HDF5 configuration ----------------------------------
+
+    if env['HDF5_DIR'] == NULL_PATH:
+        try:
+            env['HDF5_DIR'] = os.environ['HDF5_DIR']
+        except KeyError:
+            Abort('HDF5 installation directory could not be found.')
+
+    env.Append(LIBPATH = [env['HDF5_DIR']+'/lib'])
+    env.Append(LIBS    = ['hdf5'])
+    env.Append(CPPPATH = [env['HDF5_DIR']+'/include'])
```

Comment:
Following comment above, now `env["HDF5_DIR"]` would be `env["ENV"]["HDF5_DIR"]`

Reply:
See above.